### ISSUE TITLE
`best_vloss` isn't restored even if restart_filename arg is specified

### ISSUE THREAD

**Issue by k-ujihara** [IBM/materials#6]
This is relative to https://github.com/IBM/materials/commit/6861477df2b12a1066a07884170778e00d5f3852.

A saved checkpoint is loaded with this line and `self.model` and `self.start_epoch` are restored when `restart_filename` arg is specified:
https://github.com/IBM/materials/blob/5728309baabc891b24545dcf0a97fb0459c9573c/smi-ted/finetune/trainers.py#L88 

However, `best_vloss` is set to `inf` even though the value was not `inf` when the checkpoint was saved.
https://github.com/IBM/materials/blob/5728309baabc891b24545dcf0a97fb0459c9573c/smi-ted/finetune/trainers.py#L92

From my understanding, `best_vloss` is needed to be restored.

**Comment by vichShir** [comment]
Thanks for noticing that. You are right, I didn't notice that `best_vloss` must be restored when the flag `save_every_epoch` is disabled and `save_ckpt` is enabled to restore the checkpoint with the lowest validation loss.

I'll update the `_save_checkpoint` method as well to save the best validation loss to restore it in `_load_checkpoint` method.

**Comment by vichShir** [comment]
@k-ujihara I have linked the pull request [Restore best_vloss in finetune](https://github.com/IBM/materials/pull/7) with this issue to add your requested enhancements.

Do you think it's ok?

**Comment by k-ujihara** [comment]
Sure.
